packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/gollm/GoLLMController.java

```diff
@@ -147,6 +238,67 @@ public ResponseEntity<TaskResponse> createModelCardTask(
 		}
 	}
 
+	@GetMapping("/configure-model")
+	@Secured(Roles.USER)
+	@Operation(summary = "Dispatch a `GoLLM Configure Model` task")
+	@ApiResponses(value = {
+			@ApiResponse(responseCode = "200", description = "Dispatched successfully", content = @Content(mediaType = "application/json", schema = @io.swagger.v3.oas.annotations.media.Schema(implementation = TaskResponse.class))),
+			@ApiResponse(responseCode = "404", description = "The provided model or document arguments are not found", content = @Content),
+			@ApiResponse(responseCode = "500", description = "There was an issue dispatching the request", content = @Content)
+	})
+	public ResponseEntity<TaskResponse> createConfigureModelTask(
+			@RequestParam(name = "model-id", required = true) final UUID modelId,
+			@RequestParam(name = "document-id", required = true) final UUID documentId){
+
+		try {
+
+			// Grab the document
+			final Optional<DocumentAsset> document = documentAssetService.getAsset(documentId);
+			if (document.isEmpty()) {
+				return ResponseEntity.notFound().build();
+			}
+
+			// make sure there is text in the document
+			if (document.get().getText() == null || document.get().getText().isEmpty()) {
+				log.warn("Document {} has no text to send", documentId);
+				return ResponseEntity.badRequest().build();
+			}
+
+			// Grab the model
+			final Optional<Model> model = modelService.getAsset(modelId);
+			if (model.isEmpty()) {
+				return ResponseEntity.notFound().build();
```

Comment:
If its helpful for the front end, instead of returning this you could throw an exception with this status code which will allow you to pass a message to the front end on what was missing (model or document) something like... `throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Document not found");`

Reply:
added better error for model card and configure model